sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/RebaseDateTime.scala

```diff
@@ -285,12 +285,12 @@ object RebaseDateTime {
   }
 
   // Loads rebasing info from an JSON file. JSON records in the files should conform to
-  // `JsonRebaseRecord`. AnyRefMap is used here instead of Scala's immutable map because
-  // it is 2 times faster in DateTimeRebaseBenchmark.
-  private[sql] def loadRebaseRecords(fileName: String): AnyRefMap[String, RebaseInfo] = {
+  // `JsonRebaseRecord`. Mutable HashMap is used here instead of AnyRefMap due to SPARK-49491.
+  private[sql] def loadRebaseRecords(fileName: String): mutable.HashMap[String, RebaseInfo] = {
```

Comment:
```suggestion
  // `JsonRebaseRecord`. HashMap is used here instead of Scala's immutable map because
  // it is 2 times faster in DateTimeRebaseBenchmark.
  private[sql] def loadRebaseRecords(fileName: String): HashMap[String, RebaseInfo] = {
```